### PR TITLE
cap log size, post state if gist fails

### LINKED ISSUE #44: Silent failure if logs are too large
If the log files are too large to fit in a gist (I think 1MB is the limit), the final step will fail and no status will be posted.

### PR DESCRIPTION
**Pull request by simonbyrne**
Fixes #44.

### CODE REVIEW

File: finalize-test.jl [hunk 1]
```diff
@@ -25,7 +27,7 @@ function start()
 
     for jobid in keys(jobdict)
         filename = joinpath(slurmoutdir, jobid)
-        files["out_$jobid"] = Dict("content" => String(read(filename)))
+        files["out_$jobid"] = Dict("content" => String(read(filename, MAX_LOG_SIZE)))
```
Comment (jakebolewski) [inline finalize-test.jl:30]: would we want to upload the tail of the log (seek back from end MAX_LOG_SIZE)?

**Comment by simonbyrne** [comment]
I'm trying it out on https://github.com/CliMA/ClimateMachine.jl/commit/420a27e8249d9d5de8a74926024651a6d332d770

File: finalize-test.jl (hunk 1 above)
Comment (simonbyrne) [reply finalize-test.jl:30]: maybe? though usually the failure occurs earlier in the log, and the rest is noise.

Comment (jakebolewski) [reply finalize-test.jl:30]: maybe I don't quite understand why a job would error and then not fail immediately

Comment (simonbyrne) [reply finalize-test.jl:30]: > can we turn off pkg verbose=true?

It is occasionally useful, as otherwise build failures are hidden.

Comment (simonbyrne) [reply finalize-test.jl:30]: i guess if there isn't a better way to do this...

**Comment by simonbyrne** [comment]
output here: https://gist.github.com/climabot/5325d3dcfbf1ff1c1e0a92570a97944f#file-out_10011931

File: finalize-test.jl (hunk 1 above)
Comment (simonbyrne) [reply finalize-test.jl:30]: https://github.com/JuliaLang/Pkg.jl/issues/1870

Comment (simonbyrne) [reply finalize-test.jl:30]: I think it's caused by instantiate, not build.

Comment (simonbyrne) [reply finalize-test.jl:30]: it's actually curl. I've just modified my `~/.curlrc` file for now to add `--silent`

**Pull request merged**